tests/test_multi_connection.c

```diff
@@ -190,12 +192,241 @@ test_new(void **state)
     pthread_barrier_destroy(&st->barrier);
 }
 
+static void
+test_sub_suspend_helper(sr_subscription_ctx_t *subscr, int suspend)
+{
+    uint32_t sub_id;
+    int ret;
+
+    if (!suspend) {
+        return;
+    }
+    sub_id = sr_subscription_get_last_sub_id(subscr);
+    ret = sr_subscription_suspend(subscr, sub_id);
+    assert_int_equal(ret, 0);
+}
+
+struct cb_data {
+    int fail;
+};
```

Comment:
We usually use the `state` with its `cb_called` counter and check that on several places throughout the test as needed. But if you do not want to rewrite it, this is okay, too.

Reply:
I would like to leave it as it is, as this gives the `backtrace` when called unexpectedly, rather than just a counter, and makes it easier to debug a race for me.